source/draw/gpu/GpuCanvas.ooc

```diff
@@ -59,6 +58,11 @@ GpuCanvasYuv420Semiplanar: class extends GpuSurface {
 		this _target y canvas draw(gpuImage y, source, destination, map)
 		this _target uv canvas draw(gpuImage uv, IntBox2D new(source leftTop / 2, source size / 2), IntBox2D new(destination leftTop / 2, destination size / 2), map)
 	}
+	draw: override func ~ImageSourceDestination (image: Image, source, destination: IntBox2D) {
```

Comment:
Are these boxes in upper left base zero or reference coordinates?
I prefer base 0 for general purpose calculations to treat textures like arrays as in compute shaders.
Reference coordinates in integers would not even be mathematically possible for odd dimensions.